webapp-django/crashstats/signature/static/signature/js/signature_report.js

```diff
@@ -334,6 +334,180 @@ $(function () {
+            MG.convert.date(finalData, 'key', '%Y%m%d%H%M%S');
+            finalData = finalData.map(function(d) {
+                return {
+                    'date': d.key,
+                    'ratio': d.values.ratio,
+                    'adu_count': d.values.adu_count,
+                    'crash_count': d.values.crash_count
+                };
+            });
+            return finalData;
+        }
+
+        // Draws the graph
+        function drawGraph(data) {
+            // TODO: sort out window resizing
+            MG.data_graphic({
+                data: data,
+                chart_type: 'point',
+                full_width: true,
+                target: '#adu-graph',
+                x_accessor: 'date',
+                y_accessor: 'ratio',
+                xax_start_at_min: true,
+                decimals: 3,
+                area: false
+            });
+        }
+
+        // Success function for JSON request
+        function onJsonLoad(data) {
+            if (data.total) {
+                drawGraph(formatData(data.hits));
+            } else {
+                $('#adu-graph').text('No results were found.');
+            }
+        }
+
+        // Error function
+        function error(jqXHR, textStatus, errorThrown) {
+            handleError(contentElt, jqXHR, textStatus, errorThrown);
+        }
+
+        function showGraph(channel) {
+            // Remove previous results and show loader.
+            contentElt.empty();
+            addLoaderToElt(contentElt);
+
+            // Make graph options from form. Mandatory params for graph:
+            // product_name, signature, start_date, end_date, channel
+            // Dates will be processed by Django; others processed here
+            var params = getParamsWithSignature();
+
+            // Channel and signature should be defined
+            var graphOptions = {
+                channel: channel,
+                signature: params.signature
+            };
+
+            // Product may be undefined
+            // If so, user will need to choose a product
+            if (params.product) {
+                graphOptions.product = params.product[0];
+            } else {
+                // Show the search form
+                var elt = $('.toggle-filters');
+                elt.removeClass('show');
+                elt.html('Hide');
+                form.show();
+                // Add a new fieldset for choosing a product
+                form.dynamicForm('newLine', {
+                    field: 'product',
+                    operator: 'has terms',
+                    value: ''
+                });
+                // Focus the value input of the new fieldset
+                $('fieldset:last-child .value').select2('open');
+                // Prompt the user to choose a product
+                contentElt.empty().text('Please select a product.');
+                return;
+            }
+
+            // Get graph panel -- this might not be necessary
```

Comment:
Yeah there's no point getting the graph panel if it's just a simple template that won't change depending on the data. You should instead include it in the global template of that page, and only query for the graph data.